Unstable LongPollingIT.shouldActivateAndCompleteJobsInTime

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? Once
- Does it block your work? Not yet
- Do we suspect that it is a real failure? Yes, it was added as regression test and wouldn't expect that it fails except we still have some issues with the job activiation.

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Expecting
  <RetriableClientFutureImpl[Incomplete]>
to be completed within 2L Seconds.
exception caught while trying to get the future result: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.assertj.core.internal.Futures.assertSucceededWithin(Futures.java:100)
	at org.assertj.core.api.AbstractCompletableFutureAssert.internalSucceedsWithin(AbstractCompletableFutureAssert.java:442)
	at org.assertj.core.api.AbstractCompletableFutureAssert.succeedsWithin(AbstractCompletableFutureAssert.java:438)
	at io.camunda.zeebe.it.clustering.LongPollingIT.shouldActivateAndCompleteJobsInTime(LongPollingIT.java:101)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
	at io.camunda.zeebe.it.clustering.LongPollingIT.shouldActivateAndCompleteJobsInTime(LongPollingIT.java:101)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be omitted

See above, related to job activation.

In the logs we see a lot of network issues with the worker :thinking: 

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
14:32:48.864 [] [ForkJoinPool-3-worker-1] INFO  io.zeebe.containers.cluster.ZeebeCluster - Starting cluster zeebe-cluster with 1 brokers, 1 gateways, 3 partitions, and a replication factor of 1
14:32:48.865 [] [testcontainers-lifecycle-2] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Creating container for image: localhost:5000/camunda/zeebe:current-test
14:32:48.963 [] [testcontainers-lifecycle-2] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Container localhost:5000/camunda/zeebe:current-test is starting: 8ee50aa9970cb2a29a9f1ba5970d659bb3fd9f9164bb2d1f4186702b8d344831
14:32:50.039 [] [grpc-default-executor-0] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:49248
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	... 1 more
14:32:50.039 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 5274 ms
14:32:54.089 [] [ducttape-2] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /flamboyant_burnell: Waiting for 120 seconds for URL: http://localhost:49633/ready (where port 49633 maps to container port 9600)
14:32:55.314 [] [grpc-default-executor-0] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:49248
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	... 1 more
14:32:55.314 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 5035 ms
14:32:56.249 [] [testcontainers-lifecycle-2] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Container localhost:5000/camunda/zeebe:current-test started in PT7.384256938S
14:32:56.250 [] [testcontainers-lifecycle-3] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Creating container for image: localhost:5000/camunda/zeebe:current-test
14:32:56.300 [] [testcontainers-lifecycle-3] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Container localhost:5000/camunda/zeebe:current-test is starting: e3309894cce855c9de342357901d50fbe6427a8637cfeef853200e15a922ff56
14:33:00.350 [] [grpc-default-executor-0] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:49248
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	... 1 more
14:33:00.350 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 4691 ms
14:33:01.445 [] [ducttape-2] INFO  io.zeebe.containers.ZeebeTopologyWaitStrategy - /affectionate_clarke: Waiting for PT2M for topology to have at least 1 brokers, 3 partitions with 1 replicas, and each partition to have a leader
14:33:01.598 [] [ducttape-3] TRACE io.zeebe.containers.ZeebeTopologyWaitStrategy - /affectionate_clarke: Topology: TopologyImpl{brokers=[], clusterSize=0, partitionsCount=0, replicationFactor=0, gatewayVersion=8.1.0-SNAPSHOT}
14:33:02.609 [] [ducttape-3] TRACE io.zeebe.containers.ZeebeTopologyWaitStrategy - /affectionate_clarke: Topology: TopologyImpl{brokers=[BrokerInfoImpl{nodeId=0, host='zeebe-broker-0', port=26501, version=8.1.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=LEADER, health=HEALTHY}, PartitionInfoImpl{partitionId=2, role=LEADER, health=HEALTHY}, PartitionInfoImpl{partitionId=3, role=LEADER, health=HEALTHY}]}], clusterSize=1, partitionsCount=3, replicationFactor=1, gatewayVersion=8.1.0-SNAPSHOT}
14:33:02.610 [] [testcontainers-lifecycle-3] INFO  docker[localhost:5000/camunda/zeebe:current-test] - Container localhost:5000/camunda/zeebe:current-test started in PT6.360767793S
14:33:03.186 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.poller - Activated 1 jobs for worker default and job type foo
14:33:03.570 [] [grpc-default-executor-0] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
14:33:03.571 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: Network closed for unknown reason, delay retry for 53 ms
14:33:03.626 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:49643
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	... 1 more
14:33:03.626 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 80 ms
14:33:03.707 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.47.0.jar:1.47.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.47.0.jar:1.47.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.47.0.jar:1.47.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:49643
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.78.Final.jar:4.1.78.Final]
	... 1 more
14:33:03.707 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 122 ms
</pre>
</details>

```
io.camunda.zeebe.it.clustering.LongPollingIT,shouldActivateAndCompleteJobsInTime
15:22:24.205 [] [main] INFO  org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
15:22:24.221 [] [main] INFO  io.zeebe.containers.cluster.ZeebeCluster - Starting cluster zeebe-cluster with 1 brokers, 1 gateways, 3 partitions, and a replication factor of 1
15:22:24.252 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
15:22:24.272 [] [testcontainers-lifecycle-0] WARN  org.testcontainers.dockerclient.DockerClientProviderStrategy - DOCKER_HOST tcp://127.0.0.1:40599 is not listening
15:22:24.854 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
15:22:24.856 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
15:22:24.877 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.DockerClientFactory - Connected to docker: 
  Server Version: 20.10.12
  API Version: 1.41
  Operating System: Fedora Linux 35 (Thirty Five)
  Total Memory: 31374 MB
15:22:24.889 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.DockerClientFactory - Checking the system...
15:22:24.890 [] [testcontainers-lifecycle-0] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
15:22:25.053 [] [testcontainers-lifecycle-0] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Creating container for image: camunda/zeebe:SNAPSHOT
15:22:25.090 [] [testcontainers-lifecycle-0] INFO  🐳 [testcontainers/ryuk:0.3.3] - Creating container for image: testcontainers/ryuk:0.3.3
15:22:25.231 [] [testcontainers-lifecycle-0] INFO  🐳 [testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 is starting: c177123776c97684443ea940c6326a4a0746b6acf69bba9cfca8a92ed56a43ad
15:22:25.749 [] [testcontainers-lifecycle-0] INFO  🐳 [testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 started in PT0.661034024S
15:22:26.015 [] [testcontainers-lifecycle-0] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Container camunda/zeebe:SNAPSHOT is starting: e5495ef6163970091d6f62f1f6531360ea2ff1e0571a151b1deb75ac9708ef14
15:22:30.670 [] [ducttape-0] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /dreamy_cartwright: Waiting for 120 seconds for URL: http://localhost:50156/ready (where port 50156 maps to container port 9600)
15:22:31.821 [] [testcontainers-lifecycle-0] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Container camunda/zeebe:SNAPSHOT started in PT6.930170137S
15:22:31.823 [] [testcontainers-lifecycle-1] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Creating container for image: camunda/zeebe:SNAPSHOT
15:22:32.249 [] [testcontainers-lifecycle-1] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Container camunda/zeebe:SNAPSHOT is starting: 49d91e0058feade1182c534c119acbf20f09e67a42a62433d894e83f7d56262b
15:22:37.512 [] [ducttape-0] INFO  io.zeebe.containers.ZeebeTopologyWaitStrategy - /hungry_booth: Waiting for PT2M for topology to have at least 1 brokers, 3 partitions with 1 replicas, and each partition to have a leader
15:22:39.034 [] [ducttape-1] TRACE io.zeebe.containers.ZeebeTopologyWaitStrategy - /hungry_booth: Topology: TopologyImpl{brokers=[BrokerInfoImpl{nodeId=0, host='zeebe-broker-0', port=26501, version=1.4.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=LEADER, health=HEALTHY}, PartitionInfoImpl{partitionId=2, role=LEADER, health=HEALTHY}, PartitionInfoImpl{partitionId=3, role=LEADER, health=HEALTHY}]}], clusterSize=1, partitionsCount=3, replicationFactor=1, gatewayVersion=1.4.0-SNAPSHOT}
15:22:39.045 [] [testcontainers-lifecycle-1] INFO  🐳 [camunda/zeebe:SNAPSHOT] - Container camunda/zeebe:SNAPSHOT started in PT7.222754696S
15:22:40.007 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.poller - Activated 1 jobs for worker default and job type foo
15:22:40.842 [] [grpc-default-executor-0] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.48.0.jar:1.48.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.48.0.jar:1.48.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
15:22:40.851 [] [grpc-default-executor-0] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: Network closed for unknown reason, delay retry for 51 ms
15:22:40.913 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.48.0.jar:1.48.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.48.0.jar:1.48.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:50158
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	... 1 more
15:22:40.914 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 80 ms
15:22:40.995 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.48.0.jar:1.48.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.48.0.jar:1.48.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:50158
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	... 1 more
15:22:40.996 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 128 ms
15:22:41.125 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.48.0.jar:1.48.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.48.0.jar:1.48.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:50158
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	... 1 more
15:22:41.126 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 196 ms
15:22:41.323 [] [grpc-default-executor-2] WARN  io.camunda.zeebe.client.job.poller - Failed to activate jobs for worker default and job type foo
io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.48.0.jar:1.48.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487) ~[grpc-stub-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.48.0.jar:1.48.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.48.0.jar:1.48.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: localhost/127.0.0.1:50158
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.unix.Socket.finishConnect(Socket.java:321) ~[netty-transport-native-unix-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:477) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[netty-transport-classes-epoll-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.79.Final.jar:4.1.79.Final]
	... 1 more
15:22:41.324 [] [grpc-default-executor-2] DEBUG io.camunda.zeebe.client.job.worker - Failed to activate jobs due to UNAVAILABLE: io exception, delay retry for 290 ms

Process finished with exit code 0
```

--- a/gateway/src/main/java/io/camunda/zeebe/gateway/Loggers.java
+++ b/gateway/src/main/java/io/camunda/zeebe/gateway/Loggers.java
@@ -13,6 +13,8 @@ import org.slf4j.LoggerFactory;
 public final class Loggers {
 
   public static final Logger GATEWAY_LOGGER = LoggerFactory.getLogger("io.camunda.zeebe.gateway");
+  public static final Logger LONG_POLLING =
+      LoggerFactory.getLogger("io.camunda.zeebe.gateway.longPolling");
   public static final Logger GATEWAY_CFG_LOGGER =
       LoggerFactory.getLogger("io.camunda.zeebe.gateway.impl.configuration");
 }

--- a/gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/InflightActivateJobsRequest.java
+++ b/gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/InflightActivateJobsRequest.java
@@ -21,7 +21,7 @@ import org.slf4j.Logger;
 
 public class InflightActivateJobsRequest {
 
-  private static final Logger LOG = Loggers.GATEWAY_LOGGER;
+  private static final Logger LOG = Loggers.LONG_POLLING;
   private final long requestId;
   private final BrokerActivateJobsRequest request;
   private final ServerStreamObserver<ActivateJobsResponse> responseObserver;
@@ -197,7 +197,7 @@ public class InflightActivateJobsRequest {
   }
 
   @Override
-  public boolean equals(Object obj) {
+  public boolean equals(final Object obj) {
     if (this == obj) {
       return true;
     }

--- a/gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java
+++ b/gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java
@@ -36,7 +36,7 @@ import org.slf4j.Logger;
 public final class LongPollingActivateJobsHandler implements ActivateJobsHandler {
 
   private static final String JOBS_AVAILABLE_TOPIC = "jobsAvailable";
-  private static final Logger LOG = Loggers.GATEWAY_LOGGER;
+  private static final Logger LOG = Loggers.LONG_POLLING;
   private static final String ERROR_MSG_ACTIVATED_EXHAUSTED =
       "Expected to activate jobs of type '%s', but no jobs available and at least one broker returned 'RESOURCE_EXHAUSTED'. Please try again later.";
 
@@ -68,7 +68,7 @@ public final class LongPollingActivateJobsHandler implements ActivateJobsHandler
   }
 
   @Override
-  public void accept(ActorControl actor) {
+  public void accept(final ActorControl actor) {
     this.actor = actor;
     activateJobsHandler.accept(actor);
     onActorStarted();

--- a/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/LongPollingIT.java
+++ b/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/LongPollingIT.java
@@ -7,98 +7,144 @@
  */
 package io.camunda.zeebe.it.clustering;
 
-import io.camunda.zeebe.client.api.response.ProcessInstanceResult;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
+import io.camunda.zeebe.gateway.Loggers;
 import io.camunda.zeebe.model.bpmn.Bpmn;
-import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.process.test.assertions.BpmnAssert;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.JobBatchIntent;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.qa.util.actuator.LoggersActuator;
 import io.camunda.zeebe.qa.util.testcontainers.ContainerLogsDumper;
 import io.camunda.zeebe.qa.util.testcontainers.ZeebeTestContainerDefaults;
+import io.camunda.zeebe.test.util.record.RecordStream;
+import io.zeebe.containers.ZeebeGatewayNode;
+import io.zeebe.containers.ZeebePort;
 import io.zeebe.containers.cluster.ZeebeCluster;
+import io.zeebe.containers.engine.ContainerEngine;
 import java.time.Duration;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.StreamSupport;
 import org.agrona.CloseHelper;
-import org.assertj.core.api.Assertions;
+import org.assertj.core.groups.Tuple;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
 import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
 
+@Testcontainers
 final class LongPollingIT {
-  private static final Logger LOGGER = LoggerFactory.getLogger(LongPollingIT.class);
-  private static final BpmnModelInstance PROCESS =
-      Bpmn.createExecutableProcess("process")
-          .startEvent()
-          .serviceTask("task", t -> t.zeebeJobType("foo"))
-          .endEvent()
-          .done();
-
-  private Network network;
-  private ZeebeCluster cluster;
-
-  @SuppressWarnings("unused")
-  @RegisterExtension
-  final ContainerLogsDumper gatewayLogsWatcher =
-      new ContainerLogsDumper(() -> cluster.getGateways(), LOGGER);
+  private final Network network = Network.newNetwork();
+  private final ZeebeCluster cluster =
+      ZeebeCluster.builder()
+          .withBrokersCount(1)
+          .withGatewaysCount(1)
+          .withPartitionsCount(1)
+          .withEmbeddedGateway(false)
+          .withImage(ZeebeTestContainerDefaults.defaultTestImage())
+          .withGatewayConfig(this::configureGateway)
+          .build();
 
   @SuppressWarnings("unused")
   @RegisterExtension
-  final ContainerLogsDumper brokerLogsWatcher =
-      new ContainerLogsDumper(() -> cluster.getBrokers(), LOGGER);
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(cluster::getNodes);
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder().withCluster(cluster).withIdlePeriod(Duration.ofSeconds(5)).build();
 
   @BeforeEach
   void beforeEach() {
-    network = Network.newNetwork();
+    // set log level for long polling to trace to have more debugging info when the test fails
+    for (final var gateway : cluster.getGateways().values()) {
+      LoggersActuator.of(gateway).set(Loggers.LONG_POLLING.getName(), Level.TRACE);
+    }
   }
 
   @AfterEach
   void afterEach() {
-    CloseHelper.quietCloseAll(cluster, network);
+    CloseHelper.quietCloseAll(network);
   }
 
   // regression test of https://github.com/camunda/zeebe/issues/9658
   @Test
-  void shouldActivateAndCompleteJobsInTime() {
+  void shouldActivateAndCompleteJobsInTime() throws InterruptedException, TimeoutException {
     // given
-    cluster =
-        ZeebeCluster.builder()
-            .withBrokersCount(1)
-            .withGatewaysCount(1)
-            .withEmbeddedGateway(false)
-            .withPartitionsCount(3)
-            .withReplicationFactor(1)
-            .withImage(ZeebeTestContainerDefaults.defaultTestImage())
-            .build();
-    cluster.start();
-    final var zeebeClient = cluster.newClientBuilder().build();
-    final var deploymentEvent =
-        zeebeClient
-            .newDeployResourceCommand()
-            .addProcessModel(PROCESS, "process.bpmn")
-            .send()
-            .join();
-    // open the worker first and cause so to run into long polling mode
-    zeebeClient
-        .newWorker()
-        .jobType("foo")
-        .handler((c, j) -> c.newCompleteCommand(j).send())
-        .timeout(Duration.ofSeconds(10))
-        .pollInterval(Duration.ofMillis(10))
-        .open();
+    final var process =
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType("foo"))
+            .endEvent()
+            .done();
 
-    // when
-    final var resultZeebeFuture =
-        zeebeClient
-            .newCreateInstanceCommand()
-            .processDefinitionKey(deploymentEvent.getProcesses().get(0).getProcessDefinitionKey())
-            .withResult()
-            .send();
+    try (final var client = engine.createClient()) {
+      final var deploymentEvent =
+          client.newDeployResourceCommand().addProcessModel(process, "process.bpmn").send().join();
 
-    // then
-    Assertions.assertThat((CompletionStage<ProcessInstanceResult>) resultZeebeFuture)
-        .succeedsWithin(2, TimeUnit.SECONDS)
-        .isNotNull();
+      // when - send the ActivateJobs request first, before the process instance is created
+      final var jobs =
+          client
+              .newActivateJobsCommand()
+              .jobType("foo")
+              .maxJobsToActivate(1)
+              .requestTimeout(Duration.ofSeconds(30))
+              .send();
+
+      // wait until the first activate command before starting the process instance to ensure long
+      // polling will come in effect
+      engine.waitForIdleState(Duration.ofSeconds(30));
+      records()
+          .withIntent(JobBatchIntent.ACTIVATE)
+          .findFirst()
+          .orElseThrow(
+              () ->
+                  new IllegalStateException(
+                      "Expected at least one job batch ACTIVATE command, but none found"));
+      client
+          .newCreateInstanceCommand()
+          .processDefinitionKey(deploymentEvent.getProcesses().get(0).getProcessDefinitionKey())
+          .requestTimeout(Duration.ofMinutes(1))
+          .send()
+          .join();
+
+      // then - ensure that we tried to activate before the job was created, and that we activated
+      // it AGAIN after it was created, without the client sending a new request
+      engine.waitForIdleState(Duration.ofSeconds(30));
+      assertThat(records())
+          .extracting(Record::getValueType, Record::getIntent)
+          .containsSubsequence(
+              Tuple.tuple(ValueType.JOB_BATCH, JobBatchIntent.ACTIVATE),
+              Tuple.tuple(ValueType.JOB, JobIntent.CREATED),
+              Tuple.tuple(ValueType.JOB_BATCH, JobBatchIntent.ACTIVATE));
+      assertThat((CompletionStage<ActivateJobsResponse>) jobs)
+          .succeedsWithin(30, TimeUnit.SECONDS)
+          .extracting(ActivateJobsResponse::getJobs)
+          .asList()
+          .hasSize(1);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private RecordStream records() {
+    return new RecordStream(
+        StreamSupport.stream(BpmnAssert.getRecordStream().records().spliterator(), false)
+            .map(r -> (Record<RecordValue>) r));
+  }
+
+  private void configureGateway(final ZeebeGatewayNode<?> gateway) {
+    gateway
+        .withEnv("ZEEBE_GATEWAY_LONGPOLLING_ENABLED", "true")
+        // https://github.com/camunda-community-hub/zeebe-test-container/issues/332
+        .addExposedPort(ZeebePort.MONITORING.getPort());
   }
 }

--- a/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/LoggersActuator.java
+++ b/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/LoggersActuator.java
@@ -1,0 +1,89 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.qa.util.actuator;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import feign.Feign;
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+import feign.Retryer;
+import feign.Target.HardCodedTarget;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import io.zeebe.containers.ZeebeNode;
+import org.slf4j.event.Level;
+
+/**
+ * Java interface for any node's loggers actuator. To instantiate this interface, you can use {@link
+ * Feign}; see {@link #of(String)} as an example.
+ *
+ * <p>You can use one of {@link #of(String)} or {@link #of(ZeebeNode)} to create a new client to use
+ * for yourself.
+ *
+ * <p>Adding a new method is simple: simply define the input/output here as you normally would, and
+ * make sure to add the correct JSON encoding headers (`Accept` for the response type,
+ * `Content-Type` if there's a body to send).
+ */
+@SuppressWarnings({"unused", "UnusedReturnValue"})
+public interface LoggersActuator {
+
+  /**
+   * Returns a {@link LoggersActuator} instance using the given node as upstream.
+   *
+   * @param node the node to connect to
+   * @return a new instance of {@link LoggersActuator}
+   */
+  static LoggersActuator of(final ZeebeNode<?> node) {
+    final var endpoint =
+        String.format("http://%s/actuator/loggers", node.getExternalMonitoringAddress());
+    return of(endpoint);
+  }
+
+  /**
+   * Returns a {@link LoggersActuator} instance using the given endpoint as upstream. The endpoint
+   * is expected to be a complete absolute URL, e.g. "http://localhost:9600/actuator/loggers".
+   *
+   * @param endpoint the actuator URL to connect to
+   * @return a new instance of {@link LoggersActuator}
+   */
+  @SuppressWarnings("JavadocLinkAsPlainText")
+  static LoggersActuator of(final String endpoint) {
+    final var target = new HardCodedTarget<>(LoggersActuator.class, endpoint);
+    return Feign.builder()
+        .encoder(new JacksonEncoder())
+        .decoder(new JacksonDecoder())
+        .retryer(Retryer.NEVER_RETRY)
+        .target(target);
+  }
+
+  /**
+   * A convenience method to set the level of a logger remotely via the Spring Boot actuator, using
+   * SLF4J levels as input to avoid any errors (e.g. typos).
+   *
+   * @param id the logger ID, e.g. io.camunda.zeebe.gateway
+   * @param level the level you wish to set
+   */
+  default void set(final String id, final Level level) {
+    set(id, new LoggerInfo(level.toString(), null));
+  }
+
+  @RequestLine("POST /{id}")
+  @Headers({"Content-Type: application/json", "Accept: application/json"})
+  void set(@Param final String id, final LoggerInfo level);
+
+  @RequestLine("POST /{id}")
+  @Headers("Accept: application/json")
+  LoggerInfo get(@Param final String id);
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  @JsonInclude(Include.NON_EMPTY)
+  record LoggerInfo(String configuredLevel, String effectiveLevel) {}
+}

--- a/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/PartitionsActuator.java
+++ b/qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/PartitionsActuator.java
@@ -28,8 +28,8 @@ import java.util.Map;
  *
  * <p>Adding a new method is simple: simply define the input/output here as you normally would, and
  * make sure to add the correct JSON encoding headers (`Accept` for the response type,
- * `Content-Type` if there's a body to send). See {@link io.zeebe.containers.clock.ZeebeClockClient}
- * for a more complete example.
+ * `Content-Type` if there's a body to send). See {@link LoggersActuator} for a more complete
+ * example.
  */
 @SuppressWarnings({"unused", "UnusedReturnValue"})
 public interface PartitionsActuator {

